keymasterd/certnotifier/impl.go

```diff
@@ -0,0 +1,142 @@
+package certnotifier
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"github.com/Symantec/Dominator/lib/log"
+	"github.com/Symantec/keymaster/proto/certmon"
+	"io"
+	"net"
+	"net/http"
+	"time"
+)
+
+const (
+	bufferLength = 16
+)
+
+func newCertNotifier(logger log.DebugLogger) *CertNotifier {
+	return &CertNotifier{
+		logger:           logger,
+		transmitChannels: make(map[chan<- transmitType]chan<- transmitType),
+	}
+}
+
+func (n *CertNotifier) publish(certType uint32, certData []byte) {
+	transmitData := transmitType{certType, certData}
+	n.mutex.Lock()
+	defer n.mutex.Unlock()
+	for ch := range n.transmitChannels {
+		select {
+		case ch <- transmitData:
+		default:
+		}
+	}
+}
+
+func (n *CertNotifier) serveHTTP(w http.ResponseWriter, req *http.Request) {
+	if req.Method != "CONNECT" {
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+	if hijacker, ok := w.(http.Hijacker); !ok {
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	} else {
```

Comment:
I think is more clear to declare the hijacker and then avoid this else block

Reply:
Will fix.